### ISSUE TITLE
test-suite: fails to find test modules

### ISSUE THREAD

**Issue by runeksvendsen** [dramforever/vscode-ghc-simple#3]
Using `vscode-ghc-simple` with a project that includes a test-suite, when I load `test/Spec.hs` in VS Code I get the error `Failed to load interface for ...` for all imports of test depenencies, including local test modules (in the `test/` folder).

Steps to reproduce:

```bash
git clone https://github.com/runeksvendsen/orderbook
cd orderbook/
stack test #build library and test dependencies (will fail for test/Orphans/OrderBook.hs)
```

open `orderbook/` folder in VS Code and open the file `test/Orphans/OrderBook.hs`.
Result: all imports of test dependencies give an error saying `Failed to load interface for ...`.

**Comment by dramforever** [comment]
Thank you for helping by trying things out and sending feedback!

It seems that we either need to figure out which target a file is in, or load all targets when starting `stack repl`. For `stack` I think the following solves it. However, I wasn't able to figure out how to do the same for `cabal`. I think I'll try harder.

Meanwhile, this is what I've found for `stack repl`:

We can get all targets names like this:

```plain
$ stack ide targets
orderbook:lib
orderbook:test:orderbook-test
```

And when we start GHCi like this:

```plain
stack repl orderbook:lib orderbook:test:orderbook-test --no-load
```

Everything works just fine.

**Comment by dramforever** [comment]
This is also a reminder for me that building before starting repl takes ages... Gotta work harder on #2.

**Comment by runeksvendsen** [comment]
Thank *you* for spending time creating a simple Haskell IDE plugin. It's really nice to run into small issues like this one, rather than being unable to build e.g. ghc-mod for the newest GHC.

**Comment by dramforever** [comment]
I've pushed a fix for `stack` workspaces, 5a3f378468d19a8b9e9cf1aeee90ecb291c07435. I still need to work on it for `cabal` workspaces, so leaving this issue open.

I'm considering whether to put it up on to the marketplace, but if you want to try it out *now* and don't feel like using the development version, I did a version bump and made [a package for version v0.0.5 available on the releases page](https://github.com/dramforever/vscode-ghc-simple/releases/tag/v0.0.5).

**Comment by runeksvendsen** [comment]
Thanks again, @dramforever! Works like a charm now 👍

**Comment by dramforever** [comment]
I don't think `cabal repl` can load multiple targets...?

**Comment by dramforever** [comment]
`cabal repl` doesn't support multiple targets.

And apparently `stack repl` supports multiple targets by putting all the flags together and hoping that they work (they do!).

Might need to do better...

**Comment by dramforever** [comment]
Sorry for the extremely delayed response. I've recently started working on supporting multi-component/multi-package projects. This should be a less hacky way of supporting multiple targets than just specifying all the targets.

If you haven't abandoned vscode-ghc-simple, please consider testing out development versions of the `vgs-bios` branch. See more information in the discussion here: <https://github.com/dramforever/vscode-ghc-simple/discussions/87>.

In short, you will need to install a development version of the `vgs-bios` branch and set up an `hie.yaml` file, with the same format as the one accepted by [hie-bios] to tell vscode-ghc-simple about your project structure. If you need help or find any problems please do consider reaching out.

[hie-bios]: https://github.com/mpickering/hie-bios#explicit-configuration

Not everything is well-supported yet, but hopefully simple multi-component or multi-package setups will work. The same file should also help out those who use ghcide. Many projects already have an `hie.yaml` file, so these will hopefully work out of the box. See the discussion page linked above for details for more information.

Thanks for considering helping out!

**Comment by dramforever** [comment]
Released in v0.2.0
